src/librustc_mir/borrow_check/conflict_errors.rs

```diff
@@ -1190,7 +1190,16 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
         );
 
         let suggestion = match tcx.sess.source_map().span_to_snippet(args_span) {
-            Ok(string) => format!("move {}", string),
+            Ok(mut string) => {
+                if string.starts_with("async ") {
```

Comment:
This is a rather error-prone way of checking this that will fail with:

```rust
async
|x| {}
```

or

```rust
async\t|x|`
```

I don't think this should be looking at the code snippet but rather trace with `self.tcx` whether this closure was an `async` one before lowering. I sadly cannot tell you off the top of my head how to do it and cannot look at the code at this very moment but maybe someone else can chim in. @estebank?

Reply:
That is true. Ideally you'd want to get the original node, but I don't think you can. An alternative that keeps operating on the string (less than ideal, as already established) would be to use `string.split_whitespace().next()` which would yield `Some("async")` for the cases we care about in a pretty foolproof way.

Reply:
String operations for these checks is not ideal, but sometimes we have no way around them unless we track more things from the AST onwards.